tests/unit/wordpress_mock.py

```diff
@@ -0,0 +1,526 @@
+        """Install WordPress on a simulated database.
+
+        Args:
+            host: database host.
+            database: database name.
+        """
+        key = self._database_identifier(host, database)
+        if key not in self._databases:
+            raise KeyError(f"Database (host={host}, database={database}) does not exist")
+        if self._databases[key] is not None:
+            raise KeyError(f"Wordpress already installed on (host={host}, database={database}).")
+        self._databases[key] = WordPressDatabaseInstanceMock(
+            builtin_options=self._builtin_wordpress_options
+        )
+
+    def is_wordpress_installed(self, host: str, database: str) -> bool:
+        """Test if WordPress is installed on the given simulated database.
+
+        Args:
+            host: database host.
+            database: database name.
+        """
+        key = self._database_identifier(host, database)
+        if key not in self._databases:
+            raise KeyError(f"Database (host={host}, database={database}) does not exist")
+        return self._databases[key] is not None
+
+    def get_wordpress_database(self, host, database) -> WordPressDatabaseInstanceMock:
+        """Get the simulated WordPress installed database.
+
+        Args:
+            host: database host.
+            database: database name.
+        """
+        key = self._database_identifier(host, database)
+        if not self.is_wordpress_installed(host, database):
+            raise KeyError(f"Wordpress isn't installed on (host={host}, database={database}).")
+        return self._databases[key]
+
+
+class MysqlConnectorMock:
+    """A mock for :py:mod:`mysql.connector`."""
+
+    Error = mysql.connector.Error
+
+    def __init__(self, wordpress_database_mock: WordpressDatabaseMock):
+        self._wordpress_database_mock = wordpress_database_mock
+
+    def connect(self, host: str, database: str, user: str, password: str, charset: str):
+        if not self._wordpress_database_mock.database_can_connect(host, database):
+            raise self.Error(
+                msg=f"Can't connect to MySQL server on '{host}:3306' (2003)",
+                errno=2003,
+            )
+        if not self._wordpress_database_mock.database_can_login(host, database, user, password):
+            raise self.Error(
+                msg=f"Access denied for user '{host}'@* (using password: *)",
+                errno=1045,
+            )
+
+        return unittest.mock.MagicMock()
+
+
+class HandlerRegistry:
+    """A utility class that can be used to collect pattern and handler pair using decorator syntax.
+
+    For example::
+        registry = HandlerRegistry()
+
+        @registry.register(match=lambda target: target.startswith("a"))
+        def handler_func(target):
+            print(target)
+
+        match, handler = registry.registered_handler[0]
+        match("abc") # => True
+        handler("abc") # => print("abc")
+    """
+
+    def __init__(self):
```

Comment:
Should this have a return type and docstring?

Reply:
Fixed, thanks!